src/Queue/AWSSQS/SqsMessageFactory.php

```diff
@@ -0,0 +1,27 @@
+<?php declare(strict_types = 1);
+
+namespace BE\QueueManagement\Queue\AWSSQS;
+
+use function array_push;
+
+class SqsMessageFactory
+{
+    /**
+     * @param array<mixed> $awsResultMessages
+     *
+     * @return SqsMessage[]
+     */
+    public static function fromAwsResultMessages(array $awsResultMessages, string $queueUrl): array
+    {
+        /**
+         * @var SqsMessage[]
+         */
+        $sqsMessages = [];
+
+        foreach ($awsResultMessages as $message) {
+            array_push($sqsMessages, new SqsMessage($message, $queueUrl));
+        }
+
+        return $sqsMessages;
```

Comment:
```suggestion
        return array_map(
        fn($message) => return new SqsMessage($message, $queueUrl),
        $awsResultMessages
        );
```
Might not be the exact code, but my point is it could be easier than using foreach

Reply:
I am not happy with this proposed change. **_foreach_** is very easy to read, **_array_map_** with callback function is on the contrary difficult to read.  What exactly this change will improve? I think there is no benefit and readability will suffer.